tests/unit/socket_test.cc

```diff
@@ -224,3 +224,29 @@ SEASTAR_TEST_CASE(socket_on_close_local_shutdown_test) {
         when_all(std::move(client), std::move(server)).discard_result().get();
     });
 }
+
+SEASTAR_TEST_CASE(socket_connect_abort_test) {
+    return seastar::async([&] {
+        bool too_late = false;
+        auto sk = make_socket();
+        auto cf = sk.connect(ipv4_addr("10.255.255.1", 12345)).then([] (auto cs) {
```

Comment:
> What is supposed to happen in this address? I assume you hope it will never respond?

Yes. 10.0.0.0/8 is reserved range for local communications, I assume that 10.255.255.* is not routable in any ... practical environments like CI instances which seems to be true

> Maybe it's better to use a really reserved address for this? E.g., something from TEST-NET-1 (192.0.2.0/24). See https://www.rfc-editor.org/rfc/rfc5735

Ah, 192.0.2.0/24 sounds much better, thanks